tracee-ebpf/tracee/tracee.bpf.c

```diff
@@ -2128,14 +2245,54 @@ int tracepoint__sched__sched_process_exec(struct bpf_raw_tracepoint_args *ctx)
     }
 
     int invoked_from_kernel = has_prefix("kworker/", get_task_parent_comm(task), 9);
+
     const char *filename = get_binprm_filename(bprm);
+
+    struct file* file = get_file_ptr_from_bprm(bprm);
+    dev_t s_dev = get_dev_from_file(file);
+    unsigned long inode_nr = get_inode_nr_from_file(file);
+
+    unsigned long arg_start = 0, arg_end = 0;
+    int argc = 0;
+
+    // bprm->mm is null at this point (set by begin_new_exec()), and task->mm is already initialized
+    struct mm_struct *mm = get_mm_from_task(task);
+
+    arg_start = get_arg_start_from_mm(mm);
+    arg_end = get_arg_end_from_mm(mm);
+    argc = get_argc_from_bprm(bprm);
+
+    // Instruction limit exceeds when adding env vars in kernels < 5.2
+    //unsigned long env_start, env_end;
+    //env_start = get_env_start_from_mm(mm);
+    //env_end = get_env_end_from_mm(mm);
+    //int envc = get_envc_from_bprm(bprm);
+
+    // Get per-cpu string buffer
+    buf_t *string_p = get_buf(STRING_BUF_IDX);
+    if (string_p == NULL)
+        return -1;
+    save_path_to_str_buf(string_p, &file->f_path);
+    u32 *off = get_buf_off(STRING_BUF_IDX);
+    if (off == NULL)
+        return -1;
+
     u64 *tags = bpf_map_lookup_elem(&params_names_map, &context.eventid);
     if (!tags) {
         return -1;
     }
 
+    // Note: Starting from kernel 5.9, there are two new interesting fields in bprm that we should consider adding:
+    // 1. struct file *executable - which can be used to get the executable name passed to an interpreter
+    // 2. fdpath - generated filename for execveat (after resolving dirfd)
+
     save_str_to_buf(submit_p, (void *)filename, DEC_ARG(0, *tags));
-    save_to_submit_buf(submit_p, &invoked_from_kernel, sizeof(int), INT_T, DEC_ARG(1, *tags));
+    save_str_to_buf(submit_p, (void *)&string_p->buf[*off], DEC_ARG(1, *tags));
+    save_args_str_arr_to_buf(submit_p, (void *)arg_start, (void *)arg_end, argc, DEC_ARG(2, *tags));
+    //save_args_str_arr_to_buf(submit_p, (void *)env_start, (void *)env_end, envc, DEC_ARG(3, *tags));
```

Comment:
why is this commented out?

Reply:
See the comment above for "too many instructions" on kernels before 5.2. 
I would like to add it in the future (already tested and working) for kernels > 5.2

Reply:
Can you just add a compile time macro check for kernel version?

Reply:
Yes, we can add such a check, but co-re...
Let's do it in a future PR